### PR TITLE
A constructor and __str__ added for sympy.core.function.Lambda

### PR DESCRIPTION
**Pull request by Qommon**
#### Brief description of what is fixed or changed
Added a contructor <del>and modified `__str__`</del> for `Lambda` as follows:
```
>>> from sympy import Lambda
>>> def func(x, y):
...     return (x + y)**2
>>> f = Lambda.from_func(func)
```

#### Release Notes

* functions
  * Added a new constructor `from_func` for class `Lambda`.

### CODE REVIEW

**Comment by sylee957** [comment]
We mostly keep our printers in one place in `sympy.printing.str`
And we already have there.
https://github.com/sympy/sympy/blob/54b108d354fe22023dccfca9a7f44c38b39a9881/sympy/printing/str.py#L211-L217

File: sympy/core/function.py [hunk 1]
```diff
@@ -1871,6 +1871,39 @@ def __new__(cls, variables, expr):
         obj.nargs = FiniteSet(len(v))
         return obj
 
+    @classmethod
+    def from_func(cls, func):
+        """
+        'Lambda' object can also be created with a function.
+
+        A simple example:
```
Comment (smichr) [inline sympy/core/function.py]: ->
``` 
Examples
=======
```

Comment (smichr) [reply sympy/core/function.py]: It seems like this can just be added to the instantiation of Lambda. You can detect a function with `if callable(foo)`.

Comment (Qommon) [reply sympy/core/function.py]: Thank you for your review!
Do you mean that I should modify `Lambda.__new__`?
I thought only 1 parameter is needed when using a `function` but 2 parameters are needed when using `variables` and `expr`. Will this cause confusion in future developing?

**Comment by Qommon** [comment]
@sylee957
> We mostly keep our printers in one place in `sympy.printing.str`
> And we already have there.

Thank you for your review!
It's my fault that I didn't check it.
I have turned the code related to `Lambda.__str__` back to what they used to be.

File: sympy/core/function.py (hunk 1 above)
Comment (Qommon) [reply sympy/core/function.py]: Should I turn the instantiation of Lambda to this?
```
class Lambda(Expr):
    def __new__(cls, variables, expr = None):
        if callable(variables):
            from .symbol import Symbol
            variables = [Symbol(i) for i in func.__code__.co_varnames]
            expr = func(*variables)
        ......
```
Or this?
```
class Lambda(Expr):
    def __new__(cls, *args):
        if len(args)==1:
            from .symbol import Symbol
            variables = [Symbol(i) for i in func.__code__.co_varnames]
            expr = func(*variables)
        else:
            variables, expr = args
        ......
```

**Comment by asmeurer** [comment]
This needs to be qualified in the documentation that it works by calling `func(*variables)`, so that it cannot work with a function that uses Python control flow (like `if` or `for`). I'm not really sure if it's worth having given that limitation, although it would be possible at some point to have a version that does support some of that, by parsing the bytecode. 

Also it needs tests.

**Comment by smichr** [comment]
I would suggest
```python
class Lambda(Expr):
    def __new__(cls, *args):
        if len(args) != 1:
            variables, expr = args
        else:
            if not callable(args[0]):
                raise ValueError('when passing 1 argument, it should be callable')
            from .symbol import Symbol
            variables = [Symbol(i) for i in func.__code__.co_varnames]
            expr = func(*variables)
```

**Comment by Qommon** [comment]
@smichr 
Thanks for your suggestion!
I have reimplement this work in #16661 and I would appreciate it if you had time to review it after Travis CI's tests.

**Pull request closed without merging**